### PR TITLE
Allow to create OCI apprepos

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

Enable the usage of OCI registries for AppRepositories: 

Screenshot from 2021-02-08 14-00-11

Note that using a CA cert is not supported as explained here: https://github.com/kubeapps/kubeapps/pull/2323#issuecomment-775029689

### Applicable issues

  - ref #2232

### CODE REVIEW

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx [hunk 1]
```diff
@@ -372,6 +395,72 @@ export function AppRepoForm(props: IAppRepoFormProps) {
           </Row>
         </div>
       </div>
+      <div className="clr-form-control">
+        <label className="clr-control-label">Repository Type</label>
+        <span className="clr-form-description">
+          Charts can be stored either in a Helm repository (e.g. ChartMuseum) or in a OCI Registry.
+          Select the one that applies.
```
Comment (absoludity) [inline dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:402]: "Select which type of repository you are adding." (though we're confusing two different types of repositories here, another option:
"Select which type of chart catalog you are adding."

**Review by absoludity — APPROVED** [review]
LGTM, just two thoughts:

1) I wonder if in the UI we should rename "AppRepository" to "AppCatalog" so that an app catalog can have either a helm repository or an OCI registry? But that's unrelated to this PR.
2) Do you think we should have this UI change behind a feature flag? (or are you happy/ok with the sync times for it to be used and us to start maintaining it as is?)

**Comment by andresmgot** [comment]
> * I wonder if in the UI we should rename "AppRepository" to "AppCatalog" so that an app catalog can have either a helm repository or an OCI registry? But that's unrelated to this PR.

I think that can be more confusing, `AppRepository` is what we have been called all this time and the meaning (from Kubeapps point of view) is the same. Also the CRD, values.yaml, controller.. all those point to app repositories.

> * Do you think we should have this UI change behind a feature flag? (or are you happy/ok with the sync times for it to be used and us to start maintaining it as is?)

I don't think so. Better to get feedback if possible (also, I may able to improve performance before we cut a release).

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx (hunk 1 above)
Comment (andresmgot) [reply dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:402]: "Select the chart storage type"?

Comment (andresmgot) [reply dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx]: I am going to merge this but let me know if you don't agree, I can open another PR for this.

**Pull request merged**